src/sentry/api/endpoints/integrations/sentry_apps/rotate_secret.py

```diff
@@ -0,0 +1,62 @@
+from django.http import Http404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.sentryapps import SentryAppBaseEndpoint
+from sentry.api.permissions import SentryPermission
+from sentry.api.serializers import serialize
+from sentry.models.apiapplication import generate_token
+from sentry.models.integrations.sentry_app import SentryApp
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.services.hybrid_cloud.user.service import user_service
+
+
+class SentryAppRotateSecretPermission(SentryPermission):
```

Comment:
@cathteng should superusers be allowed to do this for other orgs on their behalf for support cases?

Reply:
fair. maybe we only let superusers with write privileges do this?